src/client/datascience/commands/activeEditorContext.ts

```diff
@@ -179,25 +185,30 @@ export class ActiveEditorContextService implements IExtensionSingleActivationSer
         }
         this.updateContextOfActiveNotebookKernel(activeEditor);
     }
-    private async updateDebugContext(activeEditor?: INotebookEditor, interpreter?: PythonEnvironment) {
-        this.canDebug.set(false).ignoreErrors();
-
-        if (activeEditor && activeEditor === this.notebookEditorProvider.activeEditor) {
-            const nb = await this.notebookProvider.getOrCreateNotebook({
-                identity: activeEditor.file,
-                resource: activeEditor.file,
-                getOnly: true
-            });
-            const interpreter = nb?.getMatchingInterpreter();
-            if (interpreter) {
-                this.canDebug
-                    .set(await this.dependencyService.areDebuggingDependenciesInstalled(interpreter))
-                    .ignoreErrors();
+    private onNotebookControllerSelected({
+        controller
+    }: {
+        notebook: NotebookDocument;
+        controller: VSCodeNotebookController;
+    }) {
+        this.updateDebugContext(controller.connection.interpreter);
+    }
+    private async updateDebugContext(interpreter?: PythonEnvironment) {
+        if (!interpreter) {
+            this.canDebug.set(false).ignoreErrors();
+        } else {
+            let flag = this.debugReadyInterpreters.get(interpreter);
```

Comment:
I'm actually a bit hesitant about this caching as is. If they don't have the right version installed they are going to install it from the terminal. Then it's going to continue to warn them. I'd like to get this in so we can test with it, but it's worth a follow up issue. Maybe the check could return a cached value and then start the full check each time.